apps/sms/services/js/conversation/conversation_service.js

```diff
@@ -0,0 +1,89 @@
+/*global ServiceMixin */
+'use strict';
+
+importScripts('../service_mixin.js');
+
+(function() {
+  var service;
+
+  var ConversationService = {
+    SERVICE_NAME: 'conversation-service',
+
+    STREAMS: ['getAllConversations', 'getMessagesForConversation'],
+    METHODS: [
+      'deleteConversations', 'deleteMessages', 'markConversationsAs',
+      'getConversationMetadata', 'getMessage', 'findConversationFromNumber'
+    ],
+
+    /**
+     * Initialize our service using the Service mixin's initService.
+     */
+    init() {
+      service = this.initService();
+    },
+
+    /**
+     * Stream that returns everything needed to display conversations in the
+     * inbox view.
+     * @returns {Stream.<ConversationSummary>}
+     */
+    getAllConversations() {},
+
+    /**
+     * Stream that returns messages in a conversation, with all suitable
+     * information to display in the conversation view.
+     * @param {Number} conversationId Conversation to recover.
+     * @returns {Stream.<MessageSummary>}
+     */
+    getMessagesForConversation(conversationId) {},
+
+    /**
+     * Returns all information related to a single conversation, to display in
+     * the conversation view.
+     * @param {Number} conversationId Conversation to recover.
+     */
+    getConversationSummary(conversationId){},
+
+    /**
+     * Deletes one or several conversations, including their messages.
+     * @param {Array.<Number>} conversationIds ID of the conversations to be
+     * deleted.
+     */
+    deleteConversations(conversationIds){},
+
+    /**
+     * Deletes one or several messages.
+     * @param {Array.<Number>} messageIds ID of the messages to be * deleted.
+     */
+    deleteMessages(messageIds){},
+
+    /**
+     * Marks a conversation as read or unread.
+     * @param {Object} param
```

Comment:
Ah, could you please share a link where you saw that? It's quite possible that my knowledge in JSDoc is outdated already in this fast pacing JS world :)